src/compat/browser_detection.ts

```diff
@@ -97,9 +97,20 @@ let isPlayStation5 = false;
   {
     isSafariMobile = true;
   } else if (
+    // the following statement check if the window.safari contains the method
+    // "pushNotification", this condition is not met when using web app from the dock
+    // on macOS, this is why we also check userAgent.
     Object.prototype.toString.call(globalScope.HTMLElement).indexOf("Constructor") >= 0 ||
     (globalScope as ISafariWindowObject).safari?.pushNotification?.toString() ===
-      "[object SafariRemoteNotification]"
+      "[object SafariRemoteNotification]" ||
+    // browsers are lying: Chrome reports both as Chrome and Safari in user
+    // agent string, So to detect Safari we have to check for the Safari string
+    // and the absence of the Chrome string
+    // eslint-disable-next-line max-len
+    // @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#which_part_of_the_user_agent_contains_the_information_you_are_looking_for
+    ((/Safari\/(\d+)/).test(navigator.userAgent) &&
+    !(/Chrome\/(\d+)/).test(navigator.userAgent) &&
+    !(/Chromium\/(\d+)/).test(navigator.userAgent))
```

Comment:
I thought we would also test for the presence of something like `Version` here, no?

Reply:
Yes we could also test for `Version` but is it redundant of testing that there is Safari but not Chrome/Chromium?

Reply:
Well we had the fear of false positives, if e.g. a `Chromiax` comes along or something
Wouldn't checking for Version be a safer bet? Or are we now potentially having false negatives?

Reply:
Samsung TV 2020 user agent is something like: "Mozilla, AppleWebkit TV Safari/537" so the current code will mistakenly detect it as Safari.
So I think you are right, we need to check for "Version/", but does it still make sense to check for "Safari" the absence of "Chrome" ?

Reply:
The more the reduced false positives and negative, the better.

I propose we check for all other platforms first and only indicate it's safari if it isn't the other ones AND has Safari AND has not chrome/chromium AND has Version?

The shaka also seems to check for `Apple` in `navigator.vendor`

Reply:
That useragent stuff is so annoying. I know they blame web developers for it, but they're making in fine their browsers worse here.